lms/djangoapps/courseware/views/views.py

```diff
@@ -1136,7 +1136,7 @@ def _downloadable_certificate_message(course, cert_downloadable_status):
                     course_id=course.id, uuid=cert_downloadable_status['uuid']
                 )
             )
-        else:
+        elif not cert_downloadable_status['download_url']:
```

Comment:
if cert_downloadable_status['download_url'] is not None it would call _downloadable_cert_data. Is that desirable because earlier it was only returning generated_data

Reply:
@adeelehsan Yes, that's what we are trying to fix here. Actually `certs_api.has_html_certificates_enabled(course)` is now always returning True regardless of course has valid PDF certificates or not and this else block was returning generating data instead of that PDF certificate URL.